### DISCUSSION TITLE
Drums midi output from hydrogen?

### DISCUSSION THREAD

**Discussion by roirodriguez** [hydrogen-music/hydrogen#1234] [Q&A]
Hi,

First of all i'm sorry if this question does not make sense, i'm a complete newbie both with midi and audio.

I'm connecting Hydrogen's midi out (tx) to midisnoop, then hit play. Midisnoop is silent. ¿Should i be getting midi output from hydrogen? I mean, ¿should i be getting drum midi events out of hydrogen or it acts just as sequencer / synthesizer with no (midi drums) output? if this last is the case, ¿what's hydrogen's tx for?

The use case would be to use the pattern library i created in hydrogen through other sequencer.

Thank you

**Comment by theGreatWhiteShark** [comment]
Ah sorry,  completely missed your question.

Yes, Hydrogen is able to send MIDI signals to other applications. There are two things required: 

1. a proper setup in Preferences > MIDI > Output. But you probably have this covered as you already see the output.
2. for each instrument within the current drumkit you can/have to set an output MIDI channel and the corresponding MIDI note. You can do so in the second last row of the Instrument Editor. Note, if the channel is set to "off" MIDI output is disabled for this instrument.

**Reply by philbee4** [reply]
Hi, 
Hydrogen is a powerful tool, I'm just trying to expand his soudfont capabilities to outside, i.e. I have exactly the same question : how can we get midi out messages from Hydrogen sequencer (or individual clic on an instrument inside a drumkit) to an external synth or vst.
I tried the last Hydrogen version (1.1.1) on Windows 8.1, deinstalled yet reinstalled, with above parameters given by theGreatWhiteShark, testing midi output with 2 programs : MidiOx (an old software which runs fine to dump midi messages from a midi device via soundcard but it doesn't "see"  Hydrogen as  a midi source, I think it's normal) and Coolsoft VirtualMidiSynth, set as midi output (Virtualmidisynth#1) in Hydrogen configuration. My midi pilot is "PortMidi" (cannot choose another, but is it the good setting ?)
Nothing happens between Hydrogen and VirtualMidiSynth when I strike on an instrument (having set a midi channel > 0 for this instrument) or play a sequence. 
I have successfuly tested Virtualmidisynth alone from an external keyboard via MidiOx (midi redirected to Virtualmidisynth#1) and from Coolsoft VirtualMidiKeyboard.
Did I missed or misunderstood something ? Is there some debug feature I could use ?
Thanks for help.
Phil
![HydrogenMidiSettings](https://user-images.githubusercontent.com/102689219/161261510-75732fef-02ac-4c76-9b23-801444e16141.png)

**Reply by philbee4** [reply]
Solved ! well, I don't know exactly how, but after deinstall my version and installed debug version v1.2.0 from the link I found at https://github.com/hydrogen-music/hydrogen/discussions/1532 (sorry, I should have read this post before writing my story), I got what I was searching for : an external sound from VirtualMidiSynth when playing a pattern or an instrument.
Great feature since - with some work - we can use external instruments inside drum patterns.
Thanks for this product !
Phil

**Reply by philbee4** [reply]
Yet another comment to share my  experience trying to synchronize an external midi device with Hydrogen : I had to use a midi bridge  to send Hydrogen midi out messages to a midi compatible instrument or vst manager, even if not listed as possible midi output by default. 
After many tries, I found the good pipe : a free (for personal use) Virtual MIDI Driver named LoopBe1, quite simple to install, and visible as midi output (from Hydrogen for instance) or input (from MidiOx [tester and mapper], Addictive Drums , or Cantabile, etc...) . With a latency ... lower that Hydrogen one's , at least using GmRockKit default library. So great thanks to Daniel Schmitt, who wrote LoopBe1. 
I am using a per instrument mapping to translate Hydrogen Instruments to Addictive Drums default map, and testing Hydrogen on this new functionnality (including "save as" instrument with midi parameter modifications - channel and notes ; maybe a new subject soon ...
